cmd/tailscale/cli/set.go

```diff
@@ -143,6 +151,7 @@ func runSet(ctx context.Context, args []string) (retErr error) {
 			PostureChecking: setArgs.postureChecking,
 		},
 	}
+	maskedPrefs.Prefs.NoStatefulFiltering.Set(!setArgs.statefulFiltering)
```

Comment:
took me a second to realize that this unconditional set of an opt.Bool doesn't matter because the real conditional part is the setFlagSet.Visit loop below which sets the set bits

Reply:
added a comment and moved things a bit to make it clearer